crates/vercel_runtime_macro/src/lib.rs

```diff
@@ -0,0 +1,126 @@
+use glob::glob;
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::AttributeArgs;
+use vercel_runtime_router::{Route, Router};
+
+use std::collections::HashMap;
+use syn::parse_macro_input;
+
+#[proc_macro_attribute]
+pub fn include_api(args: TokenStream, stream: TokenStream) -> TokenStream {
+    let args = parse_macro_input!(args as AttributeArgs);
+    let mut args_map: HashMap<String, String> = HashMap::new();
+
+    args.iter().for_each(|arg| {
+        if let syn::NestedMeta::Meta(syn::Meta::NameValue(syn::MetaNameValue {
+            path,
+            lit: syn::Lit::Str(lit_str),
+            ..
+        })) = arg
+        {
+            if let Some(key) = path.get_ident() {
+                args_map.insert(key.to_string(), lit_str.value());
+            }
+        }
+    });
+
+    let prefix = std::env::var("VERCEL_RUST_EXPERIMENTAL_MACRO_PREFIX").unwrap_or("".to_string());
+
+    let glob_pattern = match args_map.get("path") {
+        Some(val) => format!("{}/{}", prefix, val),
+        _ => {
+            println!("include_api: Missing `path` argument, defaulting to `../api/**/*.rs`");
+            format!("{}api/**/[!index]*.rs", &prefix)
+        }
+    };
+
+    let input: syn::ItemFn = syn::parse(stream).unwrap();
+
+    let raw_routes = glob(&glob_pattern)
+        .expect("Failed to read glob pattern")
+        .filter_map(|e| e.ok())
+        .map(|raw_path| raw_path.to_str().unwrap().to_owned())
+        .collect::<Vec<_>>();
+
+    let raw_routes = raw_routes
```

Comment:
**question:** why not have this map attached to the previous call chain? I vaguely remember there was a reason but I can't remember what it was